assignments/array-methods.js

```diff
@@ -56,28 +56,50 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 // ==== Challenge 1: Use .forEach() ====
 // The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
 let fullName = [];
-console.log(fullName);
+// get full names into fullName array by grabbing and combing their first and last names for their running bibs then push it into a new array. 
+// let splitNames = runners.forEach(function(name) {
+// fullName.push`${name.first_name} ${name.last_name}`;
+// });
+let splitNames = runners.forEach(name => fullName.push`${name.first_name} ${name.last_name}`);
+console.log(fullName)
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
-let allCaps = [];
+// let allCaps = [];
+// set first names to caps because director is power hungry.
+let allCaps = runners.map((peeps) => {
+    return peeps.first_name.toUpperCase();
+  });
 console.log(allCaps); 
 
 // ==== Challenge 3: Use .filter() ====
-// The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
-let largeShirts = [];
+// The large shirts won't be available for the event due to an ordering issue. Get a list of runners with large sized shirts so they can choose a different size. 
+// Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
+let largeShirts = runners.filter(shirtSize => {return shirtSize.shirt_size === "L"});
 console.log(largeShirts);
 
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
-let ticketPriceTotal = [];
+let ticketPriceTotal = runners.reduce((donationTotal, donationAmount) => {
+    return donationTotal + donationAmount.donation;
+  }, 0);
 console.log(ticketPriceTotal);
 
 // ==== Challenge 5: Be Creative ====
-// Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to create and then solve 3 unique problems using one or many of the array methods listed above.
+// Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  
+// Try to create and then solve 3 unique problems using one or many of the array methods listed above.
 
 // Problem 1
+let lastNamesSorted = [];
+    let filtLastNamesAlph = runners.filter(sname => lastNames.push`${name.last_name}`)
```

Comment:
remember, filter should return a boolean. this probably should be a forEach call?